backpack/utils/conv_transpose.py

```diff
@@ -33,6 +34,56 @@ def get_convtranspose3d_weight_gradient_factors(input, grad_out, module):
     return X, dE_dY
 
 
+def extract_weight_diagonal(module, input, grad_output, N):
+    """
+    input must be the unfolded input to the convolution (see unfold_func)
+    and grad_output the backpropagated gradient
+    """
+
+    def get_output_shape(input, module):
+        return module(input).shape
+
+    out_channels = module.weight.shape[0]
+    in_channels = module.weight.shape[1]
+    k = module.weight.shape[2:]
+
+    M = input.shape[0]
+    output_shape = get_output_shape(module.input0, module)
```

Comment:
`get_output_shape` is only used once, let's get rid of this function and inline it